tcell test cannot run with TERM=""

➜ termdash/tcell/tcell git:(5430f4d) TERM= go test
--- FAIL: TestNewTerminalColorMode (0.01s)
--- FAIL: TestNewTerminalColorMode/default_options (0.01s)
tcell_test.go:52: newTerminal => unexpected error:
exit status 1
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x5bf350]

goroutine 157 [running]:
testing.tRunner.func1.1(0x5f3180, 0x83b370)
/usr/lib/google-golang/src/testing/testing.go:941 +0x3d0
testing.tRunner.func1(0xc0001e6ea0)
/usr/lib/google-golang/src/testing/testing.go:944 +0x3f9
panic(0x5f3180, 0x83b370)
/usr/lib/google-golang/src/runtime/panic.go:967 +0x15d
github.com/mum4k/termdash/terminal/tcell.TestNewTerminalColorMode.func1(0xc0001e6ea0)
/usr/local/google/home/yaseenh/go/src/github.com/mum4k/termdash/terminal/tcell/tcell_test.go:56 +0xc0
testing.tRunner(0xc0001e6ea0, 0xc0001d2f30)
/usr/lib/google-golang/src/testing/testing.go:992 +0xdc
created by testing.(*T).Run
/usr/lib/google-golang/src/testing/testing.go:1043 +0x357
exit status 2
FAIL github.com/mum4k/termdash/terminal/tcell 0.040s

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,4 +17,4 @@ after_success:
         - ./internal/scripts/coverage.sh
 env:
   global:
-  - CGO_ENABLED=0
+        - CGO_ENABLED=0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [0.12.0] - 10-Apr-2020
 
+### Fixed
+
+- the `tcell` unit test can now pass in headless mode (when TERM="") which
+  happens under bazel.
+
 ### Added
 
 - Migrating to [Go modules](https://blog.golang.org/using-go-modules).

--- a/terminal/tcell/tcell.go
+++ b/terminal/tcell/tcell.go
@@ -16,6 +16,7 @@ package tcell
 
 import (
 	"context"
+	"fmt"
 	"image"
 
 	"github.com/gdamore/tcell"
@@ -79,11 +80,14 @@ type Terminal struct {
 	clearStyle *cell.Options
 }
 
+// tcellNewScreen can be overridden from tests.
+var tcellNewScreen = tcell.NewScreen
+
 // newTerminal creates the terminal and applies the options.
 func newTerminal(opts ...Option) (*Terminal, error) {
-	screen, err := tcell.NewScreen()
+	screen, err := tcellNewScreen()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("tcell.NewScreen => %v", err)
 	}
 
 	t := &Terminal{

--- a/terminal/tcell/tcell_test.go
+++ b/terminal/tcell/tcell_test.go
@@ -17,6 +17,7 @@ package tcell
 import (
 	"testing"
 
+	"github.com/gdamore/tcell"
 	"github.com/kylelemons/godebug/pretty"
 	"github.com/mum4k/termdash/cell"
 	"github.com/mum4k/termdash/terminal/terminalapi"
@@ -45,11 +46,13 @@ func TestNewTerminalColorMode(t *testing.T) {
 		},
 	}
 
+	tcellNewScreen = func() (tcell.Screen, error) { return nil, nil }
 	for _, tc := range tests {
 		t.Run(tc.desc, func(t *testing.T) {
 			got, err := newTerminal(tc.opts...)
 			if err != nil {
 				t.Errorf("newTerminal => unexpected error:\n%v", err)
+				return
 			}
 
 			// Ignore these fields.
@@ -96,11 +99,13 @@ func TestNewTerminalClearStyle(t *testing.T) {
 		},
 	}
 
+	tcellNewScreen = func() (tcell.Screen, error) { return nil, nil }
 	for _, tc := range tests {
 		t.Run(tc.desc, func(t *testing.T) {
 			got, err := newTerminal(tc.opts...)
 			if err != nil {
 				t.Errorf("newTerminal => unexpected error:\n%v", err)
+				return
 			}
 
 			// Ignore these fields.

